File: src/main/kotlin/chapter5/java/PreferencesView.java
package chapter5.java;

public class PreferencesView extends View {

    private final UserPreferences preferences;
    private final GreetingPicker greetingPicker = new GreetingPicker();
    private final LocalPicker localPicker = new LocalPicker();
    private final CurrencyPicker currencyPicker = new CurrencyPicker();

    public PreferencesView(UserPreferences preferences) {
        this.preferences = preferences;
    }

    public void show() {
        greetingPicker.setGreeting(preferences.getGreeting());
        localPicker.setLocale(preferences.getLocale());
        currencyPicker.setCurrency(preferences.getCurrency());
        super.show();
    }

    protected void onGreetingChange() {
        preferences.setGreeting(greetingPicker.getGreeting());
    }

    protected void onLocaleChange() {
        preferences.setLocale(localPicker.getLocale());
    }

    protected void onCurrencyChange() {
        preferences.setCurrency(currencyPicker.getCurrency());
    }

}

File: src/main/kotlin/chapter5/kotlin/PreferencesView.kt
package chapter5.kotlin

class PreferencesView(
    private var preferences: UserPreferences
): View() {

    private val greetingPicker = GreetingPicker()
    private val localPicker = LocalPicker()
    private val currencyPicker = CurrencyPicker()

    fun showModal(
        preferences: UserPreferences
    ): UserPreferences {
        greetingPicker.greeting = preferences.greeting
        localPicker.locale = preferences.locale
        currencyPicker.currency = preferences.currency
        show()
        return UserPreferences(
            greeting = greetingPicker.greeting,
            locale = localPicker.locale,
            currency = currencyPicker.currency
        )
    }

}